lib/coinbase/address/external_address.rb

```diff
@@ -0,0 +1,169 @@
+      if context_model.unstakeable_balance.empty? || context_model.unstakeable_balance.nil?
+        context_model.unstakeable_balance = '0'
+      end
+      if context_model.claimable_balance.empty? || context_model.claimable_balance.nil?
+        context_model.claimable_balance = '0'
+      end
+
+      {
+        stakeable_balance: Asset.to_atomic_amount(BigDecimal(context_model.stakeable_balance), asset_id),
+        unstakeable_balance: Asset.to_atomic_amount(BigDecimal(context_model.unstakeable_balance), asset_id),
+        claimable_balance: Asset.to_atomic_amount(BigDecimal(context_model.claimable_balance), asset_id)
+      }
+    end
+
+    # Retreives the stakeable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the stakeable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The stakeable balance
+    def get_stakeable_balance(asset_id, mode: :default, options: {})
+      get_staking_balances(asset_id, mode: mode, options: options)[:stakeable_balance]
+    end
+
+    # Retreives the unstakeable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the unstakeable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The unstakeable balance
+    def get_unstakeable_balance(asset_id, mode: :default, options: {})
+      get_staking_balances(asset_id, mode: mode, options: options)[:unstakeable_balance]
+    end
+
+    # Retreives the claimable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the claimable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The claimable balance
+    def get_claimable_balance(asset_id, mode: :default, options: {})
+      get_staking_balances(asset_id, mode: mode, options: options)[:claimable_balance]
+    end
+
+    private
+
+    def validate_can_stake!(amount, asset_id, mode, options)
+      stakeable_balance = get_stakeable_balance(asset_id, mode: mode, options: options)
+
+      raise InsufficientFundsError.new(amount, stakeable_balance) unless stakeable_balance >= amount
+    end
+
+    def validate_can_unstake!(amount, asset_id, mode, options)
+      unstakeable_balance = get_unstakeable_balance(asset_id, mode: mode, options: options)
+
+      raise InsufficientFundsError.new(amount, unstakeable_balance) unless unstakeable_balance >= amount
+    end
+
+    def validate_can_claim_rewards!(amount, asset_id, mode, options)
+      claimable_balance = get_claimable_balance(asset_id, mode: mode, options: options)
+
+      raise InsufficientFundsError.new(amount, claimable_balance) unless claimable_balance >= amount
```

Comment:
I think the check here should be to see whether `claimable_balance > 0`

Reply:
👍 Yes Ill add a test & update!

Reply:
We should check that the amounts are more than 0 in general otherwise they are `noop`s